Web/PathPieces.hs

```diff
@@ -20,10 +28,58 @@ import Data.Time (Day)
 import Control.Exception (assert)
 import Text.Read (readMaybe)
 
+-- | Instances of the 'PathPiece' typeclass can be converted to and from "path
+--   pieces" of URLs.
+--   <https://www.yesodweb.com/book/routing-and-handlers#routing-and-handlers_types_of_pieces In Yesod this typeclass is used to convert to and from route pieces>.
+--
+--   In the URL @"https://example.com/path/True/7"@ there are three path
+--   pieces: @"path"@, @\"True\"@, @"7"@.  These can be converted respectively into
+--   the 'Data.Text.Text', 'Bool' and 'Int' types like so:
+--
+--   > > fromPathPiece "path" :: Maybe Text
+--   > Just "path"
+--   > > fromPathPiece "True" :: Maybe Bool
+--   > Just True
+--   > > fromPathPiece "7" :: Maybe Int
+--   > Just 7
+--
+--   The return type of 'fromPathPiece' is a 'Maybe' to account for that the
+--   conversion may fail:
+--
+--   > > fromPathPiece "seven" :: Maybe Int
+--   > Nothing
+--
+-- | The 'toPathPiece' function produces URL path pieces can can be used to
```

Comment:
I believe this is supposed to be below as docs on the `toPathPiece` method itself.

Reply:
@snoyberg   I was torn between placing it above the functions themselves or at the top at the typeclass.  That's why the `|` character ended up there, haha!  I've pushed two new commits that move the docs down to each function `toPathPiece` and `fromPathPiece`.

Please let me know what do you think of the current version.